--- alpine-2.00/pith/ldap.c~	2008-04-08 00:58:40.000000000 +0200
+++ alpine-2.00/pith/ldap.c	2012-12-07 08:16:50.304560842 +0100
@@ -778,7 +778,7 @@
 	 */
 	switch(args){
 	  case 0:
-	    snprintf(filter, sizeof(filter), filt_format);
+	    snprintf(filter, sizeof(filter), "%s", filt_format);
 	    break;
 	  case 1:
 	    snprintf(filter, sizeof(filter), filt_format, scp);
--- alpine-2.00/imap/src/osdep/unix/flocklnx.c~	2008-06-04 20:18:34.000000000 +0200
+++ alpine-2.00/imap/src/osdep/unix/flocklnx.c	2012-12-07 08:13:57.761233591 +0100
@@ -57,7 +57,7 @@
     case ENOLCK:		/* lock table is full */
       sprintf (tmp,"File locking failure: %s",strerror (errno));
       mm_log (tmp,WARN);	/* give the user a warning of what happened */
-      if (!logged++) syslog (LOG_ERR,tmp);
+      if (!logged++) syslog (LOG_ERR,"File locking failure: %s",strerror (errno));
 				/* return failure if non-blocking lock */
       if (op & LOCK_NB) return -1;
       sleep (5);		/* slow down in case it loops */
--- alpine-2.00/alpine/mailcmd.c~	2011-02-08 05:34:16.000000000 +0100
+++ alpine-2.00/alpine/mailcmd.c	2012-12-07 08:18:38.574557022 +0100
@@ -7963,7 +7963,7 @@
 
 	      case 13 :			/* Match my addresses */
 		me++;
-		snprintf(sstring, sizeof(sstring), not ? _(dont_match_me) : _(match_me));
+		snprintf(sstring, sizeof(sstring), "%s", not ? _(dont_match_me) : _(match_me));
 		continue;
 
 	      case 14 :			/* Subject: default */
